### DISCUSSION TITLE
Rhino user attribute to iModel category workflow

### DISCUSSION THREAD

**Discussion by eap23** [iTwin/itwinjs-core#1611] [Q&A]
Hi,

I am following the _Bentley Accredited iTwin Developer--Associate Level_ course. In the sample _MicroStation_ file `House_Model.dgn` a new item type _SmartDevice_ was defined:
![grafik](https://user-images.githubusercontent.com/26528531/121378176-40145c00-c943-11eb-826a-c72920904db0.png)
The property `SmartDeviceId` is used to link federated data to the actual geometric object.
Following this principle I want to define a similar custom id in a _Rhino_ file. As stated here https://communities.bentley.com/products/digital-twin-cloud-services/itwin-services/w/synchronization-wiki/47595/supported-applications-and-file-formats the _Connection Type_ for "Rhino (.3dm)" files is _MicroStation_.
What would I have to define in the _Rhino_ file to have it translated to a _Category_ of it's own by the _MicroStation_ connector (similar to `DgnCustomItemTypes_HouseSchema.SmartDevice` in the sample above)?

**Comment by jffmarker** [comment]
To help clarify, are you interested in having, for example, Rhino user data get converted to item types by MicroStation, and thus it would then come across to the iModel as an object property? @c-macdonald @mgooding

**Reply by eap23** [reply]
I was thinking of using _Attribute User Text_ attache to the object in _Rhino_ to define the attributes: 
![grafik](https://user-images.githubusercontent.com/26528531/121476521-13098d00-c9c7-11eb-8101-f864ee498086.png)
![grafik](https://user-images.githubusercontent.com/26528531/121476677-40eed180-c9c7-11eb-801a-1221044fec5c.png)

If _User Text_ gets converted to item type and the attributes defined to that item's properties then the resulting _iModel_ should have the same informations as the sample DGN file, I guess.

**Reply by jffmarker** [reply]
I think ultimately that's going to need enhancements on the Rhino file I/O that MicroStation provides. I'll try to find out how they handle feature requests like this.

**Reply by jffmarker** [reply]
The MicroStation team recommends requesting Rhino file I/O enhancements on https://microstation.ideas.aha.io/

**Reply by eap23** [reply]
Thanks for checking with MicroStation team!
Are the original GUIDs from _Rhino_ persisted in the converted _iModel_?

**Reply by eap23** [reply]
@jffmarker posted at https://microstation.ideas.aha.io/ideas/MSR-I-680.

**Reply by jffmarker** [reply]
According to the MicroStation team, the initial DGN conversion does not retain the Rhino IDs, so therefore they cannot be persisted to the iModel either.
